pkg/volume/iscsi/iscsi_util.go

```diff
@@ -448,3 +468,56 @@ func removeDuplicate(s []string) []string {
 	s = s[:len(m)]
 	return s
 }
+
+func parseIscsiadmShow(output string) (map[string]string, error) {
+	params := make(map[string]string)
+	slice := strings.Split(output, "\n")
+	for _, line := range slice {
+		if !strings.HasPrefix(line, "iface") || strings.Contains(line, "<empty>") {
+			continue
+		}
+		iface := strings.Fields(line)
+		if len(iface) != 3 || iface[1] != "=" {
+			return nil, fmt.Errorf("Error invalid iface setting: %v", iface)
+		}
+		params[iface[0]] = iface[2]
+	}
+	return params, nil
+}
+
+func cloneIface(b iscsiDiskMounter, tp string) error {
+	var lastErr error
+	newIface := tp + ":" + b.volName
+	// create new iface
+	out, err := b.plugin.execCommand("iscsiadm", []string{"-m", "iface", "-I", newIface, "-o", "new"})
```

Comment:
I would rather create the new iface once pre-configured iface records are displayed without error.